Events Created By/Interacted with By UserID

staldcunha opened the pull request:
The controller takes in the userID and displays the events they created, or interacted with in three different json objects.

AarshShah9 commented on backend/controllers/event.controller.ts:
```diff
@@ -328,6 +328,82 @@ export const getEventById = async (
+      },
+    });
+
+    const eventsInterested = await prisma.event.findMany({
```
One thing I just thought of was what if they click both? Cause right now on the front-end they can heart it (to say they are interested) but also click 'going.' Should we limit it to one or the other... or both

We might have to think of the best way to approach this:
@jcharlpretorius @jtran217 @JamesRobert20

staldcunha replied:
The heart makes it simple for users to get notified of the event details. In my opinion, I would stick with the heart, to make the experience simple for the user. I can always change it up if the team sees it differently.

jcharlpretorius replied:
I think it makes sense to limit it to a single response per event. 

Right now the db model is set up so a UserEventResponse record is uniquely identified by the combination of UserId and EventId. Changing it would be trivial if we wanted to, but I don't think it makes sense to

AarshShah9 reviewed (commented):
Checkout comments please. Mostly looks like clean code though

staldcunha commented on backend/controllers/event.controller.ts:
```diff
@@ -328,6 +328,82 @@ export const getEventById = async (
+      );
+      throw notFoundError;
+    }
+    res.status(200).json(eventsCreatedByUser);
```
@jcharlpretorius brought up aa good point of condensing the json into one single response during dev night. Implementing that now!

staldcunha commented:
I have updated the following things:

- sends on json object for all events related to the user. 
- sends an empty object if user has either created no events or if user is not interested in any events. 
- only added "interested" status to the participation status.
- adjusted the seed script for testing.
